mani_skill/envs/tasks/tabletop/poke_cube.py

```diff
@@ -0,0 +1,192 @@
+        super().__init__(*args, robot_uids=robot_uids, **kwargs)
+
+    @property
+    def _default_sensor_configs(self):
+        pose = sapien_utils.look_at(eye=[0.3, 0, 0.6], target=[-0.1, 0, 0.1])
+        return [CameraConfig("base_camera", pose, 128, 128, np.pi / 2, 0.01, 100)]
+
+    @property
+    def _default_human_render_camera_configs(self):
+        pose = sapien_utils.look_at([0.6, 0.7, 0.6], [0.2, 0.2, 0.35])
+        return CameraConfig("render_camera", pose, 512, 512, 1, 0.01, 100)
+
+    def _load_scene(self, options: dict):
+        self.table_scene = TableSceneBuilder(
+            self, robot_init_qpos_noise=self.robot_init_qpos_noise
+        )
+        self.table_scene.build()
+
+        self.cube = actors.build_cube(
+            self.scene,
+            half_size=self.cube_half_size,
+            color=[1, 0, 0, 1],
+            name="cube",
+            body_type="dynamic",
+        )
+
+        self.peg = actors.build_twocolor_peg(
+            self.scene,
+            length=self.peg_half_length,
+            width=self.peg_half_width,
+            color_1=np.array([176, 14, 14, 255]) / 255,
+            color_2=np.array([12, 42, 160, 255]) / 255,
+            name="peg",
+            body_type="dynamic",
+        )
+
+        self.goal_region = actors.build_red_white_target(
+            self.scene,
+            radius=self.goal_radius,
+            thickness=1e-5,
+            name="goal_region",
+            add_collision=False,
+            body_type="kinematic",
+        )
+
+        self.peg_head_offsets = Pose.create_from_pq(p=[self.peg_half_length, 0, 0])
+
+    @property
+    def peg_head_pos(self):
+        return self.peg.pose.p + self.peg_head_offsets.p
+
+    @property
+    def peg_head_pose(self):
+        return self.peg.pose * self.peg_head_offsets
+
+    def _initialize_episode(self, env_idx: torch.Tensor, options: dict):
+        with torch.device(self.device):
+            b = len(env_idx)
+            self.table_scene.initialize(env_idx)
+            # initialize the peg
+            peg_xyz = torch.zeros((b, 3))
+            peg_xyz = torch.rand((b, 3)) * 0.2 - 0.1
+            peg_xyz[..., 2] = self.peg_half_width
+            peg_q = [1, 0, 0, 0]
+            peg_pose = Pose.create_from_pq(p=peg_xyz, q=peg_q)
+            self.peg.set_pose(peg_pose)
+            # initialize the cube
+            cube_xyz = torch.zeros((b, 3))
+            cube_xyz = torch.rand((b, 3)) * 0.2 - 0.1
+            cube_xyz[..., 0] = peg_xyz[..., 0] + self.peg_half_length + 0.1
+            cube_xyz[..., 2] = self.cube_half_size
+            cube_q = [1, 0, 0, 0]
+            cube_pose = Pose.create_from_pq(p=cube_xyz, q=cube_q)
+            self.cube.set_pose(cube_pose)
+            # initialize the goal region
+            goal_region_xyz = cube_xyz + torch.tensor([0.05 + self.goal_radius, 0, 0])
+            goal_region_xyz[..., 2] = 1e-3
+            goal_region_q = euler2quat(0, np.pi / 2, 0)
```

Comment:
can you make the task slightly more difficult by randomizing the quaternion for the cube? This seems too easy. I expect it to be still solvable with a bit of z-axis randomization

Reply:
Add z-axis randomization of the cube, rotate in the range [-pi/6, pi/6]